### PR TITLE
Fix issue with corrupted zipfiles downloaded from stations without data

### PR DESCRIPTION
**Pull request by corviday**
Address [issue 62](https://github.com/pacificclimate/pdp/issues/62):

- Update required `pydap.handlers.pcic` version from 0.0.4 to 0.0.8. 
- Add tests for data download 
- Add user documentation of behaviour when downloading from a station without data. 
- Update test of spreadsheet functionality.

### CODE REVIEW

File: tests/test_functional.py [hunk 1]
```diff
@@ -285,6 +285,49 @@ def test_legend_caching(pcic_data_portal):
...
+              'download-timeseries': 'Timeseries'
+              }
+
+    
+    req = Request.blank(base_url + urlencode(params))
+    
+    req.cookies['beaker.session.id'] = authorized_session_id
+     
+    resp = req.get_response(pcic_data_portal)
+    print resp.status
```
Comment (jameshiebert) [inline tests/test_functional.py]: Remove the print statement for production. Also get in the habit of using the print function rather than the print statement.

File: doc/source/pcds.rst [hunk 2]
```diff
@@ -63,6 +63,7 @@ There are a few nuances to the aggregated data download that work the way the us
 
 1. Even if you zoom in the the map and limit your field of view, there may be selected stations that extend beyond the field of view. Stations are selected based on your entire set of filters, and are completely unaffected by your view on the map.
 2. When downloading data, all variables for a selected station are included in the download *regardless of whether you filter by variable*. This is slightly different behaviour than is exhibited by the time selection, so it's a point that warrants attention.
+3. When downloading timeseries data, if you have selected stations by drawing a polygon but have not filtered those stations by date range or meterological variable, your selection may include stations with no data available. Files for these stations will still appear in your downloaded archive, but they will not contain any observations, only metadata and headings.
```
Comment (jameshiebert) [inline doc/source/pcds.rst]: Good. That should be a sufficient explanation.

File: requirements.txt [hunk 3]
```diff
@@ -23,7 +23,7 @@ ply==3.4
 psycopg2==2.6
 pydap-pdp==3.2.4
 pydap.handlers.csv==0.3
-pydap.handlers.pcic==0.0.4
+pydap.handlers.pcic==0.0.8
```
Comment (jameshiebert) [inline requirements.txt]: Check.

File: tests/test_functional.py [hunk 4]
```diff
@@ -112,9 +112,9 @@ def test_xls_response(pcic_data_portal, authorized_session_id):
     for i, val in enumerate(['Precip_Climatology', 127.128, 91.2249, 77.3313, 46.2816, 39.6803, 33.1902, 22.3557, 22.448, 38.5025, 72.3281, 144.159, 121.008]):
         assert obs.cell_value(i, 0) == val
     # Check a few of the global attributes
-    assert attributes.cell_value(4, 2) == '-123.283333' #longitude
-    assert attributes.cell_value(1, 2) == '1010066' # station_id
-    assert attributes.cell_value(7, 2) == 'ACTIVE PASS' # station name
+    assert attributes.cell_value(5, 2) == '-123.283333' #longitude
```
Comment (jameshiebert) [inline tests/test_functional.py]: What's the cause of the shift here?

File: tests/test_functional.py [hunk 5]
```diff
@@ -285,6 +285,49 @@ def test_legend_caching(pcic_data_portal):
...
+    resp = req.get_response(pcic_data_portal)
+    print resp.status
+    assert resp.status == '200 OK'
+
+    t = TemporaryFile()
+    t.write(resp.body)
+    z = ZipFile(t, 'r')
+    assert z.testzip() == None  
+ 
+
```
Comment (jameshiebert) [inline tests/test_functional.py]: Only leave 2 blank lines between top level functions.

File: tests/test_functional.py [hunk 6]
```diff
@@ -285,6 +285,49 @@ def test_legend_caching(pcic_data_portal):
     req.if_modified_since = pre_load_time
     resp = req.get_response(pcic_data_portal)
     assert resp.status.startswith('200')
+    
+
+@pytest.mark.crmpdb
+@pytest.mark.parametrize('polygon', ['single station polygon',
```
Comment (jameshiebert) [inline tests/test_functional.py]: Cool, great tests! This will be nice to have all of these weird corner cases in the test suite.

File: tests/test_functional.py (hunk 4 above)
Comment (corviday) [inline tests/test_functional.py]: The addition of "elevation" to NC_GLOBAL, made [here](https://github.com/pacificclimate/pydap.handlers.pcic/commit/f1204cb5e05a988617fbcdb02095212ef88ebf38). The elevation value shows up at the top of the file and pushes everything else down a row.

File: tests/test_functional.py (hunk 6 above)
Comment (corviday) [inline tests/test_functional.py]: They're a little slow, I could see us wanting to remove some of them, or assign them to a skippable pytest mark.

File: tests/test_functional.py (hunk 1 above)
Comment (corviday) [inline tests/test_functional.py]: Whoops, thanks.

**Comment by jameshiebert** [comment]
Perfect!
